### PR TITLE
Extend build-and-collect-pgo-profiles to automatically save profiles for testing

### PR DESCRIPTION
**Pull request by briannafan**
#### 04478f06ac46e9c5fea55604f1cc85d6ba5731a6
<pre>
Extend build-and-collect-pgo-profiles to automatically save profiles for testing
<a href="https://bugs.webkit.org/show_bug.cgi?id=242416">https://bugs.webkit.org/show_bug.cgi?id=242416</a>
rdar://96560886

Reviewed by Dewei Zhu.

Change -b flag to -o, taking an output directory to store profiles.

* Tools/Scripts/build-and-collect-pgo-profiles:

Canonical link: <a href="https://commits.webkit.org/252250@main">https://commits.webkit.org/252250@main</a>
</pre>

### CODE REVIEW

File: Tools/Scripts/build-and-collect-pgo-profiles [hunk 1]
```diff
@@ -125,5 +127,14 @@ $SPTH/pgo-profile combine --jetstream "$BASE/jetstream" --speedometer "$BASE/spe
 
 $SPTH/pgo-profile compress --input "$BASE/output" --output "$BASE/Internal/WebKit/WebKitAdditions/Profiling/"
 
-echo "Done! Find your profiles in $BASE/Internal/WebKit/WebKitAdditions/Profiling/"
-echo "To check these in, do: 'cp -r $BASE/Internal/ ../Internal/'"
+if [[ -n $SAVEPROF ]] ; then
+    mv "$BASE" "/Volumes/Data/$SAVEPROF-BenchmarkProfiles"
```
Comment (dewei-zhu) [inline Tools/Scripts/build-and-collect-pgo-profiles]: I don't think we can assume this path exists `/Volumes/Data/`

File: Tools/Scripts/build-and-collect-pgo-profiles [hunk 2]
```diff
@@ -125,5 +130,10 @@ $SPTH/pgo-profile combine --jetstream "$BASE/jetstream" --speedometer "$BASE/spe
 
 $SPTH/pgo-profile compress --input "$BASE/output" --output "$BASE/Internal/WebKit/WebKitAdditions/Profiling/"
 
+
```
Comment (dewei-zhu) [inline Tools/Scripts/build-and-collect-pgo-profiles]: Remove this line?

File: Tools/Scripts/build-and-collect-pgo-profiles [hunk 3]
```diff
@@ -125,5 +130,10 @@ $SPTH/pgo-profile combine --jetstream "$BASE/jetstream" --speedometer "$BASE/spe
...
 echo "Done! Find your profiles in $BASE/Internal/WebKit/WebKitAdditions/Profiling/"
 echo "To check these in, do: 'cp -r $BASE/Internal/ ../Internal/'"
+
+
+
+
```
Comment (dewei-zhu) [inline Tools/Scripts/build-and-collect-pgo-profiles]: No extra blank lines?

File: Tools/Scripts/build-and-collect-pgo-profiles [hunk 4]
```diff
@@ -110,12 +115,12 @@ fi
...
 
-$SPTH/run-benchmark "${jsargs[@]}"
-$SPTH/pgo-profile merge "$BASE/jetstream"
-
 $SPTH/run-benchmark "${spargs[@]}"
 $SPTH/pgo-profile merge "$BASE/speedometer"
 
+$SPTH/run-benchmark "${jsargs[@]}"
+$SPTH/pgo-profile merge "$BASE/jetstream"
+
```
Comment (dewei-zhu) [inline Tools/Scripts/build-and-collect-pgo-profiles]: Do we want to swap the order in this change?

Comment (briannafan) [reply Tools/Scripts/build-and-collect-pgo-profiles]: I did it for testing reasons as Speedometer is faster, I don't think swapping them will create any meaningful changes right?

**Review by dewei-zhu — APPROVED** [review]
r=me

**Review by dewei-zhu — APPROVED** [review]
r=me

**Pull request merged**